### PR TITLE
Add "migraphx" pipeline to rocmlir-driver

### PR DESCRIPTION
**Pull request by djramic**

### CODE REVIEW

File: mlir/test/fusion/pr-e2e/mixr-bcast-add.e2e.mlir [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// RUN: rocmlir-opt -migraphx-to-tosa %s | rocmlir-driver -host-pipeline highlevel | rocmlir-gen -ph -print-results -rand none - | rocmlir-driver -arch %arch -c  | mlir-cpu-runner -O2 --shared-libs=%linalg_test_lib_dir/libmlir_rocm_runtime%shlibext,%conv_validation_wrapper_library_dir/libconv-validation-wrappers%shlibext,%linalg_test_lib_dir/libmlir_runner_utils%shlibext,%linalg_test_lib_dir/libmlir_float16_utils%shlibext --entry-point-result=void | FileCheck %s
+// RUN: rocmlir-driver -kernel-pipeline migraphx %s | rocmlir-driver -host-pipeline highlevel | rocmlir-gen -ph -print-results -rand none - | rocmlir-driver -arch %arch -c  | mlir-cpu-runner -O2 --shared-libs=%linalg_test_lib_dir/libmlir_rocm_runtime%shlibext,%conv_validation_wrapper_library_dir/libconv-validation-wrappers%shlibext,%linalg_test_lib_dir/libmlir_runner_utils%shlibext,%linalg_test_lib_dir/libmlir_float16_utils%shlibext --entry-point-result=void | FileCheck %s
```
Comment (manupak) [inline mlir/test/fusion/pr-e2e/mixr-bcast-add.e2e.mlir]: What happens if we do `rocmlir-driver -kernel-pipeline migraphx,highlevel here` ? and remove `rocmlir-driver -host-pipeline highlevel`

File: mlir/tools/rocmlir-driver/rocmlir-driver.cpp [hunk 2]
```diff
@@ -262,7 +265,8 @@ static LogicalResult runMLIRPasses(ModuleOp &module,
     }
   }
 
-  bool isHighLevel = hostPipelineSet.contains("highlevel");
+  bool isHighLevel = hostPipelineSet.contains("highlevel") ||
+                     kernelPipelineSet.contains("highlevel");
```
Comment (manupak) [inline mlir/tools/rocmlir-driver/rocmlir-driver.cpp:269]: kernelPipelineSet.contains("highlevel") i dont think this is tested ? I have suggested a way to do that in the other comment.

**Review by manupak — COMMENTED** [review]
broadly looks good to me. 
just a nit around testing kernel pipeline having highlevel which is something (_I think_) we have done before -- that is not to say it not right.

File: mlir/test/fusion/pr-e2e/mixr-bcast-add.e2e.mlir (hunk 1 above)
Comment (djramic) [reply mlir/test/fusion/pr-e2e/mixr-bcast-add.e2e.mlir]: We are getting the same output. I will change that in the next commit.

**Review by manupak — APPROVED** [review]
LGTM!

**Pull request merged**